src/components/experiment/Experiment.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react';
+import ListGroup from 'react-bootstrap/ListGroup';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Toast from 'react-bootstrap/Toast';
+import Container from 'react-bootstrap/Container';
+import ExperimentItem from './ExperimentItem';
+
+const generateItems = (experimentUrls: Array<string>) => {
+  const items: Array<JSX.Element> = [];
+  experimentUrls.forEach(url => {
+    items.push(<ExperimentItem url={url} key={url} />);
+  });
+
+  return items;
+};
+
+const Experiment: React.FC = () => {
+  const [hasError, setErrors] = useState(false);
+  const [experimentUrls, setExperimentUrls] = useState([]);
+  const [isUploading, setUploading] = useState(false);
+  const [showToast, setShowToast] = useState(false);
+  const [responseMessage, setResponseMessage] = useState('');
+  const [selectedFile, setSelectedFile] = useState({ file: null });
+
+  useEffect(() => {
+    async function fetchExperimentUrls(): Promise<void> {
+      try {
+        const response = await fetch(
+          `${process.env.REACT_APP_BACKEND_ADDRESS}/experiment`
+        );
+        const body = await response.json();
+        setExperimentUrls(body.files);
+      } catch (err) {
+        setErrors(err);
+      }
+    }
+
+    if (!isUploading) fetchExperimentUrls();
+  }, [isUploading]);
+
+  const onFileSelect = async (e: any) => {
+    setSelectedFile({ file: e.target.files[0] });
+  };
+
+  const handleUploadClick = async () => {
+    const formData = new FormData();
+    if (selectedFile && selectedFile.file) {
+      formData.append('file', (selectedFile.file as unknown) as File);
+    }
+    setUploading(true);
+    const response = await fetch(
+      `${process.env.REACT_APP_BACKEND_ADDRESS}/experiment`,
+      {
+        method: 'POST',
+        body: formData,
+      }
+    );
+    const body = await response.json();
+    setResponseMessage(body.message);
+    setUploading(false);
+    setShowToast(true);
+    setSelectedFile({ file: null });
+  };
+
+  return (
+    <Container>
+      {!hasError && (
+        <Row>
+          <Col>
+            <div className="input-group">
```

Comment:
We should add a little bit of bottom margin to the upload bar so that there's some whitespace between it and the list of files.

Reply:
Changed